prolog/list_util.pl

```diff
@@ -129,20 +117,21 @@
 %  Xs = Take, Take = [] ;
 %  Xs = Take, Take = [_G3810].
 %  ==
-split_at(N,Xs,Take,Rest) :-
-    split_at_(Xs,N,Take,Rest).
+split_at(N, Xs, Take, Rest) :-
+    (  maplist(var, [N,Take])
```

Comment:
Minor issue: maybe use `var(N), var(Take)` instead of maplist/2 since meta-predicates are slightly slower to execute.